meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -175,43 +183,82 @@ IMAGE_CMD_resinos-img () {
     #
 
     dd if=/dev/zero of=${RESIN_RAW_IMG} bs=1024 count=0 seek=${RESIN_RAW_IMG_SIZE}
-    parted -s ${RESIN_RAW_IMG} mklabel msdos
+
+    if [ "${PARTITION_TABLE_TYPE}" != "msdos" ] && [ "${PARTITION_TABLE_TYPE}" != "gpt" ]; then
+        bbfatal "Unrecognized partition table: ${PARTITION_TABLE_TYPE}"
+    fi
+
+    parted --align=none ${RESIN_RAW_IMG} mklabel ${PARTITION_TABLE_TYPE}
+
+    device_specific_configuration
 
     # resin-boot
-    START=${RESIN_IMAGE_ALIGNMENT}
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
+        OPTS="primary fat16"
+    elif [ "${PARTITION_TABLE_TYPE}" = "gpt" ]; then
+        OPTS="resin-boot"
+    fi
```

Comment:
I guess we could merge this if/else with the one above, i.e. add an else that prints the "Unrecognized partition table" message.

Reply:
I have split the if statements for readability, but I guess I can merge them.